### ISSUE TITLE
Setting no validate or validationSchema clears out imperative error state on onBlur/onChange

### ISSUE THREAD

**Issue by inooid** [jaredpalmer/formik#834]
## Current Behavior
Whenever you setup a Formik form and supply no `validate` or `validationSchema`, but you call `setErrors`/`setFieldError` as if it's coming from an API, it will clear out the entire `errors` object from Formik whenever you change or blur ANY field.

![formik-validation2](https://user-images.githubusercontent.com/1291263/44349700-e4c90480-a49d-11e8-8c7f-a1fcc3606570.gif)

## Steps to Reproduce
See codesandbox for more info

## Expected behavior
I would say whenever you apply no `validate` or `validationSchema`, it should only clear out the error of the field you touched, not the entire form.

## Suggested solution(s)
I guess `onBlur` and `onChange` validation should only clear out the field it is actually touching/changing.

## Additional context
Note: The Formik version of `0.11.11` does not clear the `errors` object on change or blur at all (which to me seems a bit more reasonable, since you explicitly set the errors yourself and there's not validation, so you'll have to manually clear them as well).

Note: I haven't tested if this behaviour also happens whenever you do have a `validate` or `validationSchema` setup.

**CodeSandbox Link**:
https://codesandbox.io/s/pykpn4386m

---
- Formik Version: 1.0.3
- React Version: 16.3.2
- TypeScript Version: -
- Browser and Version: Chrome Version 68.0.3440.106 (Official Build) (64-bit)
- OS: OS X 10.13.6
- Node Version: -
- Package Manager and Version: -

**Comment by inooid** [comment]
I don't think this is stale. I am curious if this behaviour is intended or whether it's a bug 🤔

**Comment by MDornian** [comment]
I am experiencing the same problem and setting the third parameter "shouldValidate" on either setFieldValue or setFieldError has not impact.

**Comment by eugenechu** [comment]
Hey guys, any updates on this?
Seems like it is still reproducible

**Comment by vtereshyn** [comment]
I get a similar issue when I try to `setFieldError` which I haven't in `validationSchema`. So I can get this error, but when I'm editing another field, an errors object contains all of the errors except error from `setFieldError`

I guess this issue because 
```
return promise
      .then((errors: FormikErrors<Values>) => {
        this.setState(prevState => {
          if (!isEqual(prevState.errors, errors)) {
            return { errors };
          }
          return null; // abort the update
        });
        return errors;
      })
      .catch(x => x);
```

Here prevState.errors and errors are different, and if they are different we return new errors object...
Maybe I'm incorrect here...

@jaredpalmer , could you please help us here?

**Comment by simon-nystrom** [comment]
@vtereshyn I'm also struggling with this specific problem atm. Can't use custom validators for other fields if I combine them in the same <Formik /> as <Field />s with validation

**Comment by simon-nystrom** [comment]
@vtereshyn how did you do that? I'm trying to combine `setFieldError` with a custom field that isn't uinsg the Formik <Field />

**Comment by vtereshyn** [comment]
@simon-nystrom , why you should use custom field without`Formik`? use your custom input with formik `Field` and pass `validate` prop with your function and inside this function just return errorMessage. Also, inside your `renderInput` function you should check if error exists. Look through examples which exist at Formik website

**Comment by vtereshyn** [comment]
@simon-nystrom , glad to help :)

**Comment by alexander2001i** [comment]
Any updates on the issue? It's still happening.

**Comment by vigneshwaran-chandrasekaran** [comment]
is this issue fixed? I am still facing this problem. I am using version 1.5.8

**Comment by vigneshwaran-chandrasekaran** [comment]
@jaredpalmer can you please help us to solve this problem

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah I have the same problem.

**Comment by thj-dk** [comment]
I'm still seeing this problem...

**Comment by ImSolitude** [comment]
Anyone managed to solve this?

**Comment by val-samonte** [comment]
I think this needs to be reopened.

**Comment by dev99problems** [comment]
@ImSolitude @val-samonte TBH I don't really think anybody gonna handle this, because **this was reproducible** in early **1.0.3 Formik** version, which **was in alfa** at that time.

 And if I'm not mistaken it was fixed in **1.1** or some newer versions.
So if you still use Formik 1.0.3 or similar alfa version maybe you should try to update it or if you still face this issue in newer versions of Formik maybe it will be more appropriate to open a new issue.

**Comment by ZahidSaeed7** [comment]
This is still NOT fixed. Any solution?

**Comment by annaloukianova** [comment]
This is still happening in v2.2.9. Using `validate` prop does cause the errors to be set correctly and that is a good way to go, for anyone googling this, but if you weren't using the validate prop before, it might cause your `onSubmit` function not to fire if `errors` is not empty. If you switch to using `validate`, make sure to return an empty object when your form is valid and clear out any custom errors.

**Comment by CalvinD3v** [comment]
> So, for someone still finding this thread, you can just set `validateOnBlur={false}` and `validateOnChange={false}` on your `Formik` component

@msnegurski what happens if one still needs that  `validateOnBlur={false}` and `validateOnChange={false}` on your `Formik` component ?

**Comment by CalvinD3v** [comment]
@inooid did you find any solution to this issue?

**Comment by inooid** [comment]
@CalvinD3v honestly can't remember what I did to solve this. There are some ideas listed above though 👍

**Comment by orionivv** [comment]
I still need to solve this problem 😐
with `validateOnChange={true}` ❗️

**Comment by dayoolacodes** [comment]
> validateOnBlur={false}

this fixed the error trigger on blur. Thanks

**Comment by kpebron** [comment]
added `validateOnBlur={false}` but not `validateOnChange={false}` worked for me with both `validate` on frontend and `actions.setFieldError` on backend

**Comment by jaxkashif34** [comment]
> So, for someone still finding this thread, you can just set `validateOnBlur={false}` and `validateOnChange={false}` on your `Formik` component

yes it works when we change other fields it'll not remove the custom error but then it'll remove your custom error on `onSubmit` and show other errors if any in other fields
